packages/utils/src/promise.ts

```diff
@@ -1,8 +1,5 @@
-export function create<T>(id: number | string): Deferred<T> {
-    // intentionally ignore below lines in test coverage, they will be overridden in promise creation
-    /* istanbul ignore next */
+export const createDeferred = <T>(id: number | string) => {
```

Comment:
```suggestion
export const createDeferred = <T, P = undefined>(id?: P) => {
```
second generic type should allow you to pass anything as argument (no need to -1)

Reply:
I fixed it like below and it works:

```javascript
export const createDeferred = <T, P = string | number | undefined>(id?: P) => {
```